flink-core/src/test/java/org/apache/flink/api/common/typeutils/SerializerTestInstance.java

```diff
@@ -62,19 +62,27 @@ protected int getLength() {
 	}
 	
 	// --------------------------------------------------------------------------------------------
-	
+
 	public void testAll() {
-		testInstantiate();
-		testGetLength();
-		testCopy();
-		testCopyIntoNewElements();
-		testCopyIntoReusedElements();
-		testSerializeIndividually();
-		testSerializeIndividuallyReusingValues();
-		testSerializeAsSequenceNoReuse();
-		testSerializeAsSequenceReusingValues();
-		testSerializedCopyIndividually();
-		testSerializedCopyAsSequence();
-		testSerializabilityAndEquals();
+		try {
```

Comment:
Do you know why these were not called before? Also, why do we need the `try-catch` around this now?

Reply:
I guess they were added later to the base class and the person that added those cases was probably not aware of existence of this class.

The try-catch is because a few of the new test cases throw exception, whereas the `testAll` doesn't. Did not want to change it in every place where the `testAll` is called. I think though in the long run we should revisit those places and actually get rid of this method at all.

Reply:
so this isn't even compiling at the moment?

Reply:
it does

Reply:
I just realized though there some test failures for different serializers, investigating...

Reply:
I think it would probably be cleaner to not catch the exception and simply declare that it throws. Which might lead to other places where we need to declare, as you said.

Reply:
I would suggest maybe replacing this with:
```
for (Method method : getClass().getMethods()) {
			if (method.getAnnotation(Test.class) == null) {
				continue;
			}
			try {
				method.invoke(this);
			}
			catch (IllegalAccessException e) {
				throw new RuntimeException("Unable to invoke test " + method.getName(), e);
			}
			catch (InvocationTargetException e) {
				sneakyThrow(e.getCause());
			}
		}
```

where sneakyThrow is defined as follows

```
private static <E extends Throwable> void sneakyThrow(Throwable e) throws E {
		throw (E) e;
	}
```